### PR TITLE
Initial mould of stories

### PR DESCRIPTION
**Pull request by syed-qure**

### CODE REVIEW

**Comment by abhijeet-qure** [comment]
@syed-qure Still unable to run this on local, getting this error now
Also, why are we using this to install tailwind, is it not installable via npm?

CleanShot 2022-08-22 at 17 42 39@2x

**Comment by abhijeet-qure** [comment]
@syed-qure can you update the source branch with `master`?

**Comment by syed-qure** [comment]
looking into the build fail issue

**Comment by abhijeet-qure** [comment]
@syed-qure Which node version are you using? i am getting this warning while doing `npm install`, this says, it was updated with latest version.

CleanShot 2022-08-23 at 16 36 23@2x


We are using `Node 14` for all the projects, lets try to stick to that, use `nvm` for node versioning

**Comment by syed-qure** [comment]
> @syed-qure Which node version are you using? i am getting this warning while doing `npm install`, this says, it was updated with latest version.
> 
> CleanShot 2022-08-23 at 16 36 23@2x
> 
> We are using `Node 14` for all the projects, lets try to stick to that, use `nvm` for node versioning

Using 16.15.0
I'll try with 14 and see if this goes away

File: postcss.config.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  plugins: {
+    tailwindcss: {},
```
Comment (abhijeet-qure) [inline postcss.config.js:3]: So, this will make sure to add the tailwind css?

**Comment by abhijeet-qure** [comment]
You will need to export components in this file, so they are accessible whenever pushed to the `npm repo`

CleanShot 2022-08-23 at 17 20 15@2x

File: src/composite/Button/__tests__/Button.test.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  const variantClassNames = variants[variant];
+  const sizeClassNames = sizes[size];
+
+  return `${defaultStylesAsClassNames} ${variantClassNames} ${sizeClassNames}`;
+};
+
+describe("Primary Button", () => {
+  const Primary = composeStory(PrimaryStory, Meta);
+
+  it("renders with default args", () => {
```
Comment (abhijeet-qure) [inline src/composite/Button/__tests__/Button.test.tsx:23]: These are basic tests, we will need to improve on it, its fine for now, we will do this later

File: src/composite/Button/constants.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { SIZES, VARIANTS } from "./model";
+
+export const variants: Record<VARIANTS, string> = {
```
Comment (abhijeet-qure) [inline src/composite/Button/constants.ts:3]: this is really good!

File: src/composite/Button/index.tsx [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import Button from "./Button";
+import IButton from "./model";
```
Comment (abhijeet-qure) [inline src/composite/Button/index.tsx:2]: Not sure, why is it named `IButton`?

File: src/composite/Button/model.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { ReactNode } from "react";
```
Comment (abhijeet-qure) [inline src/composite/Button/model.ts:1]: In this file, Where did you see this format of exporting components?

File: .eslintrc.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  "parserOptions": {
+    "ecmaFeatures": {
+      "jsx": false
+    },
+    "ecmaVersion": 12,
+    "sourceType": "module"
+  },
+  "plugins": ["react", "@typescript-eslint"],
+  "rules": {
+    "react/prop-types": 0
```
Comment (abhijeet-qure) [inline .eslintrc.json:22]: @syed-qure this needed to be removed

**Review by abhijeet-qure — APPROVED** [review]
@syed-qure approving this PR for now, will merge it

**Pull request merged**